tests/ndimage/footprint_filter_test.py

```diff
@@ -0,0 +1,192 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+import numpy as np
+import scipp as sc
+import scipy.ndimage
+from scipp.ndimage import (generic_filter, maximum_filter, median_filter,
+                           minimum_filter, percentile_filter, rank_filter)
+import pytest
+
+filters = (generic_filter, maximum_filter, median_filter, minimum_filter,
+           percentile_filter, rank_filter)
+
+
+def make_histogram2d():
+    da = sc.DataArray(sc.array(dims=['x', 'y'], values=np.random.rand(100, 100)))
+    da.coords['x'] = sc.linspace('x', 0.0, 10.0, num=101, unit='mm')
+    da.coords['y'] = sc.linspace('y', 0.0, 50.0, num=101, unit='mm')
+    return da
+
+
+def make_data2d():
+    da = sc.DataArray(sc.array(dims=['x', 'y'], values=np.random.rand(100, 100)))
+    da.coords['x'] = sc.arange('x', 0.0, 10.0, 0.1, unit='mm')
+    da.coords['y'] = sc.arange('y', 0.0, 50.0, 0.5, unit='mm')
+    return da
+
+
+@pytest.fixture(scope="module",
+                params=[
+                    generic_filter, maximum_filter, median_filter, minimum_filter,
+                    percentile_filter, rank_filter
+                ])
+def filter_func(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=[maximum_filter, median_filter, minimum_filter])
+def simple_filter_func(request):
+    return request.param
+
+
+def test_raises_TypeError_when_output_arg_given(filter_func):
+    da = make_histogram2d()
+    with pytest.raises(TypeError):
+        filter_func(da, size=2, output=da.values)
+
+
+def test_raises_VariancesError_when_data_has_variances(filter_func):
+    da = make_histogram2d()
+    da.variances = da.values
+    with pytest.raises(sc.VariancesError):
+        filter_func(da, size=2)
+
+
+def test_raises_ValueError_when_data_has_masks(filter_func):
+    da = make_histogram2d()
+    da.masks['mask'] = da.coords['x'] == da.coords['x']
+    with pytest.raises(ValueError):
+        filter_func(da, size=2)
+
+
+def test_raises_TypeError_if_size_given_as_array_like(filter_func):
+    da = make_histogram2d()
+    with pytest.raises(TypeError):
+        filter_func(da, size=[2, 3])
+
+
+def test_raises_TypeError_if_origin_given_as_array_like(filter_func):
+    da = make_histogram2d()
+    with pytest.raises(TypeError):
+        filter_func(da, size=[2, 3])
+
+
+def test_raises_TypeError_if_size_is_not_integral(filter_func):
+    da = make_histogram2d()
+    with pytest.raises(TypeError):
+        filter_func(da, size=1.5)
+    with pytest.raises(TypeError):
+        filter_func(da, size=[2.5, 3.5])
```

Comment:
Should this use a dict instead? Otherwise it fails because of the list, not because of floats. (Same below)

Reply:
Good catch!

Reply:
You changed the test below but not this one. Was that on purpose?

Reply:
No. Done now.